File: tests/testdata/codegen/aws-resource-options-5.16.2-pp/dotnet/aws-resource-options-5.16.2.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var provider = new Aws.Provider("provider", new()
    {
        Region = "us-west-2",
    });

    var bucket1 = new Aws.S3.Bucket("bucket1", new()
    {
    }, new CustomResourceOptions
    {
        Provider = provider,
        DependsOn =
        {
            provider,
        },
        Protect = true,
        IgnoreChanges =
        {
            "bucket",
            "lifecycleRules[0]",
        },
    });

});

File: tests/testdata/codegen/aws-resource-options-5.16.2-pp/python/aws-resource-options-5.16.2.py
import pulumi
import pulumi_aws as aws

provider = aws.Provider("provider", region="us-west-2")
bucket1 = aws.s3.Bucket("bucket1", opts = pulumi.ResourceOptions(provider=provider,
    depends_on=[provider],
    protect=True,
    ignore_changes=[
        "bucket",
        "lifecycleRules[0]",
    ]))
